### PR TITLE
Add extra version of py-jsonschema

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
Needs new package: py-pysistent

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 1]
```diff
@@ -12,9 +12,15 @@ class PyJsonschema(PythonPackage):
     homepage = "http://github.com/Julian/jsonschema"
     url      = "https://pypi.io/packages/source/j/jsonschema/jsonschema-2.6.0.tar.gz"
 
+    version('3.0.1', sha256='0c0a81564f181de3212efa2d17de1910f8732fa1b71c42266d983cd74304e20d')
     version('2.6.0', sha256='6ff5f3180870836cae40f06fa10419f557208175f13ad7bc26caa77beb1f6e02')
     version('2.5.1', sha256='36673ac378feed3daa5956276a829699056523d7961027911f064b52255ead41')
 
+    depends_on('python@2.6:2.7.999,3.4:')
+    depends_on('python@2.7:2.7.999,3.5:', when='@2.6.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]: Where did you find this? Also, they should be `type=('build', 'run')`

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 2]
```diff
@@ -12,9 +12,15 @@ class PyJsonschema(PythonPackage):
...
     depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]: Is this needed at run-time?

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 3]
```diff
@@ -12,9 +12,15 @@ class PyJsonschema(PythonPackage):
...
     depends_on('py-vcversioner', type=('build', 'run'))
     depends_on('py-functools32', when="^python@2.7.0:2.7.999", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]:
```suggestion
    depends_on('py-functools32', when="^python@:2", type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 4]
```diff
@@ -12,9 +12,15 @@ class PyJsonschema(PythonPackage):
...
     url      = "https://pypi.io/packages/source/j/jsonschema/jsonschema-2.6.0.tar.gz"
 
+    version('3.0.1', sha256='0c0a81564f181de3212efa2d17de1910f8732fa1b71c42266d983cd74304e20d')
     version('2.6.0', sha256='6ff5f3180870836cae40f06fa10419f557208175f13ad7bc26caa77beb1f6e02')
     version('2.5.1', sha256='36673ac378feed3daa5956276a829699056523d7961027911f064b52255ead41')
 
+    depends_on('python@2.6:2.7.999,3.4:')
+    depends_on('python@2.7:2.7.999,3.5:', when='@2.6.0:')
     depends_on('py-setuptools', type='build')
     depends_on('py-vcversioner', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]: I don't think this dep is needed anymore.

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 5]
```diff
@@ -12,9 +12,15 @@ class PyJsonschema(PythonPackage):
...
     depends_on('py-functools32', when="^python@2.7.0:2.7.999", type=('build', 'run'))
+    depends_on('py-attrs@17.4.0:', when='@3.0.1:', type=('build', 'run'))
+    depends_on('py-pyrsistent@0.14.0:', when='@3.0.1:', type=('build', 'run'))
+    depends_on('py-six@1.11.0:', when='@3.0.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]: Missing build dep on `py-setuptools-scm`

File: var/spack/repos/builtin/packages/py-jsonschema/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-jsonschema/package.py]: https://github.com/Julian/jsonschema/commit/2387d08e05bdd9d3a65a01eb9d0c97be8c43a6bd . Will fix the type.

**Comment by iarspider** [comment]
I think `py-vcversioner` is not needed at runtime?

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 6]
```diff
@@ -12,9 +12,19 @@ class PyJsonschema(PythonPackage):
...
 
-    depends_on('py-setuptools', type='build')
-    depends_on('py-vcversioner', type=('build', 'run'))
-    depends_on('py-functools32', when="^python@2.7.0:2.7.999", type=('build', 'run'))
+    depends_on('python@2.6:2.7.999,3.4:', type=('build', 'run'))
+    depends_on('python@2.7:2.7.999,3.5:', when='@3.0.0:', type=('build', 'run'))
+
+    depends_on('py-setuptools', type='build', when='@:2.6.0')
+    depends_on('py-setuptools', type=('build', 'run'), when='@3:')
+    depends_on('py-vcversioner', type=('build', 'run'), when='@:2.6.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]:
```suggestion
    depends_on('py-vcversioner', type='build', when='@:2')
```

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 7]
```diff
@@ -12,9 +12,19 @@ class PyJsonschema(PythonPackage):
...
     version('2.6.0', sha256='6ff5f3180870836cae40f06fa10419f557208175f13ad7bc26caa77beb1f6e02')
     version('2.5.1', sha256='36673ac378feed3daa5956276a829699056523d7961027911f064b52255ead41')
 
-    depends_on('py-setuptools', type='build')
-    depends_on('py-vcversioner', type=('build', 'run'))
-    depends_on('py-functools32', when="^python@2.7.0:2.7.999", type=('build', 'run'))
+    depends_on('python@2.6:2.7.999,3.4:', type=('build', 'run'))
+    depends_on('python@2.7:2.7.999,3.5:', when='@3.0.0:', type=('build', 'run'))
+
+    depends_on('py-setuptools', type='build', when='@:2.6.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]:
```suggestion
    depends_on('py-setuptools', type='build', when='@:2')
```

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 8]
```diff
@@ -12,9 +12,19 @@ class PyJsonschema(PythonPackage):
...
+    depends_on('py-setuptools', type=('build', 'run'), when='@3:')
+    depends_on('py-vcversioner', type=('build', 'run'), when='@:2.6.0')
+    depends_on('py-setuptools-scm', type='build', when='@3:')
+
+    depends_on('py-functools32', when="^python@:2", type=('build', 'run'))
+    depends_on('py-attrs@17.4.0:', when='@3.0.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]:
```suggestion
    depends_on('py-attrs@17.4.0:', when='@3:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 9]
```diff
@@ -12,9 +12,19 @@ class PyJsonschema(PythonPackage):
...
+    depends_on('py-pyrsistent@0.14.0:', when='@3.0.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]:
```suggestion
    depends_on('py-pyrsistent@0.14.0:', when='@3:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 10]
```diff
@@ -12,9 +12,19 @@ class PyJsonschema(PythonPackage):
...
+    depends_on('py-six@1.11.0:', when='@3.0.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]:
```suggestion
    depends_on('py-six@1.11.0:', when='@3:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 11]
```diff
@@ -12,9 +12,19 @@ class PyJsonschema(PythonPackage):
...
     url      = "https://pypi.io/packages/source/j/jsonschema/jsonschema-2.6.0.tar.gz"
 
+    version('3.0.1', sha256='0c0a81564f181de3212efa2d17de1910f8732fa1b71c42266d983cd74304e20d')
     version('2.6.0', sha256='6ff5f3180870836cae40f06fa10419f557208175f13ad7bc26caa77beb1f6e02')
     version('2.5.1', sha256='36673ac378feed3daa5956276a829699056523d7961027911f064b52255ead41')
 
-    depends_on('py-setuptools', type='build')
-    depends_on('py-vcversioner', type=('build', 'run'))
-    depends_on('py-functools32', when="^python@2.7.0:2.7.999", type=('build', 'run'))
+    depends_on('python@2.6:2.7.999,3.4:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]:
```suggestion
    depends_on('python@2.6:2.8,3.4:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-jsonschema/package.py [hunk 12]
```diff
@@ -12,9 +12,19 @@ class PyJsonschema(PythonPackage):
...
+    depends_on('python@2.7:2.7.999,3.5:', when='@3.0.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jsonschema/package.py]:
```suggestion
    depends_on('python@2.7:2.8,3.5:', when='@3:', type=('build', 'run'))
```

**Pull request merged**